doc/limit/SONiC Limit user session.md

```diff
@@ -0,0 +1,222 @@
+# SONiC Limit user login session
+
+# Table of Contents
+- [Table of Contents](#table-of-contents)
+- [About this Manual](#about-this-manual)
+- [1 Functional Requirements](#1-functional-requirement)
+  * [1.1 Limit the login session per user/group/system](#11-limit-the-login-session-per-user/group/system)
+  * [1.3 Default limitation](#13-default-limitation)
+- [2 Configuration and Management Requirements](#2-configuration-and-management-requirements)
+  * [2.1 SONiC CLI](#21-sonic-cli)
+  * [2.2 Config DB](#22-config-db)
+- [3 Design](#design)
+  * [3.1 Login Limit Implementation](#31-login-limit-implementation)
+  * [3.2 Default login session limitation Implementation](#32-default-login-session-limitation-Implementation)
+  * [3.3 ConfigDB Schema](#33-configdb-schema)
+  * [3.4 CLI](#34-cli)
+- [4 Error handling](#error-handling)
+- [5 Serviceability and Debug](#serviceability-and-debug)
+- [6 Unit Test](#unit-test)
+  * [6.1 Login session enable/disable test](#61-login-session-enable-disable-test)
+  * [6.2 Max login session test](#62-max-login-session-test)
+- [8 References](#references)
+
+
+# About this Manual
+This document provides a detailed description on the new features for:
+ - Limit the number of logins per user/group/system.
+ - Default limit user login session by device information.
+ - Design a scalable framework of config command and ConfigDB to support more resource type, for example limit user CPU/memory by cgroup and systemd user.slices.
+
+## SONiC memory issue sloved by this feature.
+ - Currenly SONiC enabled OOM killer, and set 2 to /proc/sys/vm/panic_on_oom, which will trigger kernel panic when OOM. This is by design to protect SONiC key process and container.
+ - A typical switch device have 4 GB memory and sonic usually will use 1.5 GB for dockers, and 500 MB for system process. so there will be 2 GB free memory for user. also sonic not enable swap for most device.
+ - When user run some command trigger OOM, SONiC will kernel panic. for example:
```

Comment:
>trigger OOM, SONiC will kernel panic [](http://example.com/codeflow?start=30&length=36)

Why OOM will lead to kernel panic? Explain OOM a little bit for background knowledge?

Reply:
Hua need add panic on OOM detail to document.